proposals/tags-on-existing-images.md

```diff
@@ -0,0 +1,33 @@
+# Proposal: Control the pushing of tags on existing images in target registry
+
+Relevant issues: [#2360](https://github.com/GoogleContainerTools/jib/issues/2360)
+
+## Motivation
+Currently Jib pushes tags for all built images, irrelevant of whether the image is already 
+present on the target registry. 
+
+As highlighted by @loosebazooka [#2360](https://github.com/GoogleContainerTools/jib/issues/2360) 
+this can be particularly annoying if some automated orchestrator is monitoring the registry
+and releases any new tags found. This issue is especially prevalent in cases where the tags are
+auto-generated by CI systems and added with every triggered build.
+
+## Proposal
+The idea is to give the user the ability to control this functionality by means of a switch/option. 
+There might be use-cases where the current behaviour is preferable, eg, where the tags follow semantic 
+versioning, or where the tags are not automatically generated. Hence, adding a switch seems to be 
+a better solution rather than changing the current behaviour.
+
+## Proposed changes
+A new system property is to be added to control this new behaviour:
+`-Djib.skipExistingImages`
+
+A new step is created with the below signature:
+`class CheckImageStep implements Callable<BuildResult>`, where its `call()` function performs a 
+call to the `RegistryClient`. This step is executed in the `StepsRunner.pushImages()` method, 
+before the `PushImageStep.makeList()` part, only if the `skipExistingImages` flag is set to true. 
+Depending on the `BuildResult` of this new step, the `PushImageStep` futures are then created (or not).
+
+The registry call used in `CheckImageStep` can be implemented in various ways. It is being 
+proposed that a new `checkManifest` function is added in `RegistryClient`. This should execute 
+a `ManifestChecker`, also a new class. The `ManifestChecker`'s `handleHttpResponseException` class
```

Comment:
I saw in your original PR you just used `RegistryClient.pullManifest()`. What's the reason you changed your mind to create a new `ManifestChecker`?

Reply:
`pullManifest` throws an exception if the manifest does not exist. If used directly in the step, some exception handling would be needed. Anyone making the same check elsewhere would need to handle the same exception, possibly repeating code. 

The `ManifestChecker` seems more inline with the current code structure, eg, `checkBlob` using `BlobChecker`. Alternatively if it seems like over engineered, the `pullManifest` can be used within the `checkManifest` in the `RegistryClient`, and exception handling is centralized in the `RegistryClient`. The latter was my original choice, however it seems much cleaner handling the exception within the `handleHttpResponseException` method rather than a singluar try-catch block.

Reply:
Alright, let's go with `ManfiestChecker`. I looked at the `pullManifest` code, and now I think we do need a way to distinguish "manifest not found" from other errors.